### PR TITLE
Fix digi legs sprites getting stuck

### LINKED ISSUE #67174: Digitigrade legs don't squish within modsuits
## Reproduction:

Digitigrade legs can wear modsuits just fine, as MODboots ignore digitigrade, 
However, MOD boots also cover legs. 
This creates an issue, as digitigrade legs will not squish when conforming to a modsuit, showing them sticking right out the side awkwardly. 

This is likely because digitigrade limb's `update_limb` procs ONLY check for uniform and suit, instead of **any** equipment the user may be wearing that covers legs.

### LINKED ISSUE #86756: Digitigrade jumpsuit can be seen through EVA suits
### Client Version:

515.1643

### Issue Summary:

When playing as a lizard with digitigrade legs and wearing an EVA suit, parts of the jumpsuit can be seen when they should logically be concealed by the EVA suit. This might affect other suit items (like firesuits maybe) as it's probably an issue with sprite overlays but I haven't tested it with them.

image
In this picture I am wearing a green jumpsuit and the knees/ankle parts can be seen sticking out.

### Round ID:

[23802](https://scrubby.melonmesa.com/round/23802)

### Test Merge Information:

probably n/a

### Reproduction Steps:

1. Be lizard with digitigrade legs
2. Put on jumpsuit with digitigrade sprite
3. Put on EVA suit
4. See the bulky legs of the digitigrade jumpsuit that should be concealed by the EVA suit

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Fixes #86756
Fixes #67174 (isn't this already fixed? Oh well now it's really fixed because I tested it)

Basically changed the random calls to `update_body_parts` with a bespoke proc which better explains what they are there to do

Adds one missing call to head items (for `HIDESNOUT`)

## Changelog

:cl: Melbert
fix: Fixed digitigrade pants sprite not updating in accordance to some leg updates
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
Actually the astute among you may notice this PR is currently doing almost nothing. I think it's fixed on master but it does seem to have some jank. I will investigate further

**Comment by MrMelbert** [comment]
Tl;dr is that #87100 caused rendering to freak out due to always calling update body

**Review by Ghommie — APPROVED** [review]
